crates/aptos/src/common/types.rs

```diff
@@ -1264,6 +1274,14 @@ impl TransactionOptions {
                 .await?
                 .into_inner();
 
+            // Check if the transaction will pass, if it doesn't then fail
+            // TODO: Add move resolver so we can explain the VM status with a proper error map
+            let status = simulated_txn.info.status();
+            if !status.is_success() {
+                let status = client.explain_vm_status(status);
+                return Err(CliError::SimulationError(status));
+            }
+
             // Take the gas used and use a headroom factor on it
             let adjusted_max_gas = adjust_gas_headroom(
```

Comment:
Looks like this function can add a 50% buffer to max gas returned from the simulation. This seems a bit arbitryrary but is fine for now I guess. In the future I think we should revisit and improve